### PR TITLE
First draft of Overview

### PR DESCRIPTION
**Pull request by korken89**
This is after the `overview_and_structure` branch PR.

Please have a read and comment. I put more or less everything that came to mind and we might need to reorganize the information a bit.

### CODE REVIEW

File: src/overview.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+[Segger JLink]: https://www.segger.com/products/debug-probes/j-link/
+
+### [Black Magic Probe]
+
+The BMP (Black Magic Probe) is different from OpenOCD, where it is hardware, commonly the same as a debug adapter, which runs an GDB server over serial inside it, so there is no need to install OpenOCD - in a sense, it is OpenOCD combined with a debug adapter in the same dongle.
+The BMP is open hardware and runs open source software inside it as well, plus using GDB with the BMP is no different to OpenOCD, down to a few commands, so now there is a plethora of adapters to use.
+
+[Black Magic Probe]: https://github.com/blacksphere/blackmagic/wiki
+
```
Comment (ryankurte) [inline src/overview.md]: Probably worth adding ARMs DAPLink as a section too.
https://os.mbed.com/docs/latest/tools/daplink.html

File: src/overview.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+### Debugger used in this book
+
+In coherence with the Discovery Book, this book uses the STM32F3Discovery platform for all examples, which has an STLink v2 debug adapter which we will use OpenOCD to talk to.
```
Comment (ryankurte) [inline src/overview.md]: I wonder whether we could do that common API docs thing which lets you select your tools (usually programming language) for each example?
ime texane/stlink is simpler than getting OpenOCD working, what we default to may need more discussion?

File: src/overview.md (hunk 1 above)
Comment (korken89) [reply src/overview.md]: Indeed, I will add it

File: src/overview.md (hunk 2 above)
Comment (korken89) [reply src/overview.md]: I mainly chose OpenOCD so we can touch on ITM and SWO (texane does not have this support).
If we don't need it I can change :)

**Comment by korken89** [comment]
I will merge for now and we can start PRs for editorial change :)

**Pull request merged**